src/librustdoc/core.rs

```diff
@@ -570,6 +587,75 @@ pub fn run_core(options: RustdocOptions) -> (clean::Crate, RenderInfo, RenderOpt
     })
 }
 
+use rustc_hir::def::Res;
+use rustc_hir::{
+    intravisit::{NestedVisitorMap, Visitor},
+    Path,
+};
+use rustc_middle::hir::map::Map;
+
+thread_local!(static DEFAULT_TYPECK: for<'tcx> fn(rustc_middle::ty::TyCtxt<'tcx>, rustc_span::def_id::LocalDefId) -> &'tcx rustc_middle::ty::TypeckTables<'tcx> = {
+    let mut providers = rustc_middle::ty::query::Providers::default();
+    rustc_typeck::provide(&mut providers);
+    providers.typeck_tables_of
+});
```

Comment:
Ideally don't refer to crate names if you can avoid it - `ty` should be imported directly, and so should `TyCtxt` and `LocalDefId`, I'm a bit surprised you're not getting internal lints.